src/JsonReader.cs

```diff
@@ -548,127 +570,162 @@ sealed class NonProperty : IJsonProperty<Unit, JsonReadResult<Unit>>
-                if (ReadPropertyValue(property15, ref reader, ref value15, ref error)) continue; if (error is not null) return Error(error);
-                if (ReadPropertyValue(property16, ref reader, ref value16, ref error)) continue; if (error is not null) return Error(error);
-
-                if (reader.IsFinalBlock)
-                    reader.Skip();
-                else if (!reader.TrySkip())
-                    throw PartialJsonNotSupportedException();
+                            DefaultUnassigned(property1,  ref values.V1);
+                            DefaultUnassigned(property2,  ref values.V2);
+                            DefaultUnassigned(property3,  ref values.V3);
+                            DefaultUnassigned(property4,  ref values.V4);
+                            DefaultUnassigned(property5,  ref values.V5);
+                            DefaultUnassigned(property6,  ref values.V6);
+                            DefaultUnassigned(property7,  ref values.V7);
+                            DefaultUnassigned(property8,  ref values.V8);
+                            DefaultUnassigned(property9,  ref values.V9);
+                            DefaultUnassigned(property10, ref values.V10);
+                            DefaultUnassigned(property11, ref values.V11);
+                            DefaultUnassigned(property12, ref values.V12);
+                            DefaultUnassigned(property13, ref values.V13);
+                            DefaultUnassigned(property14, ref values.V14);
+                            DefaultUnassigned(property15, ref values.V15);
+                            DefaultUnassigned(property16, ref values.V16);
+
+                            return (values.V1, values.V2, values.V3,
+                                    values.V4, values.V5, values.V6,
+                                    values.V7, values.V8, values.V9,
+                                    values.V10, values.V11, values.V12,
+                                    values.V13, values.V14, values.V15,
+                                    values.V16) is ((true, var v1), (true, var v2), (true, var v3),
+                                                    (true, var v4), (true, var v5), (true, var v6),
+                                                    (true, var v7), (true, var v8), (true, var v9),
+                                                    (true, var v10), (true, var v11), (true, var v12),
+                                                    (true, var v13), (true, var v14), (true, var v15),
+                                                    (true, var v16))
+                                 ? Value(projector(v1, v2, v3, v4, v5, v6, v7, v8, v9, v10, v11, v12, v13, v14, v15, v16))
+                                 : Error("Invalid JSON object.");
+
+                        case ObjectReadStateMachine.ReadResult.PropertyName:
+                            static bool TrySetPropertyIndex<TValue>(int index,
```

Comment:
nit: Consider renaming `TValue` to simply `T`.

Reply:
Done in 0c09682f373f4e7535f978dacba5d24e8187ba9f

Reply:
I think this got undone by c2099a95eb80da5ff4053f931fad1fefd0eae74c?

Reply:
Good catch and fixed with 810da9fe796174023a451507aa7c77f251360b2b.